src/solvers/flattening/arrays.cpp

```diff
@@ -110,76 +109,64 @@ void arrayst::collect_arrays(const exprt &a)
 
   if(a.id()==ID_with)
   {
-    DATA_INVARIANT(
```

Comment:
Looking closer it seems some of them are handled by conversion functions, but there are still some that check the array members have the same type which I think should stay.